### ISSUE TITLE
Cannot run occ command after upgrading from 8.0.6 to 8.1.2RC with encryption enabled

### ISSUE THREAD

**Issue by jvillafanez** [owncloud/core#18706]
### Steps to reproduce
1. Install OC 8.0.6 from packages
2. Enable encryption
3. Upgrade to 8.1.2RC1 through packages

### Expected behaviour
Upgrade is done correctly

### Actual behaviour
Execution bit for the occ command isn't set

### Server configuration
**Operating system**: ubuntu 14.04

**Web server:** apache 2.4.7

**Database:** mysql

**PHP version:**5.5.9

**ownCloud version:** 8.1.2RC1

**Updated from an older ownCloud or fresh install:** upgrade from 8.0.6

### Notes
* Without encryption, the execution bit is set up properly
* Specific error message is "sudo: /var/www/owncloud/occ: command not found". Checked that the file is there but without the execution bit set

@jnweiger

**Comment by jvillafanez** [comment]
It also happens from 8.1.1 to 8.1.2RC1 with encryption

**Comment by cmonteroluque** [comment]
@schiesbn @PVince81 can I please ask you to have a quick look at this?

**Comment by jvillafanez** [comment]
In addition, some files, in particular the occ file, is owned by root, which prevents running the occ command properly because it should be owned by the apache user. This happens with and without encryption

**Comment by PVince81** [comment]
If the execution bit is not set then you cannot run the command.

The bug is likely that the execution bit is missing.
CC @jnweiger

**Comment by PVince81** [comment]
I just downloaded the tarball and it's missing execution permissions:
```
vincent@vvortex ~/temp/owncloud-8.1.2RC1
  % ll
total 116K
drwxr-xr-x 1 vincent users  524 Aug 30 10:43 3rdparty
drwxr-xr-x 1 vincent users  484 Aug 30 10:44 apps
-rw-r--r-- 1 vincent users  477 Aug 30 10:43 AUTHORS
drwxr-xr-x 1 vincent users   78 Aug 30 10:43 config
-rw-r--r-- 1 vincent users 2.5K Aug 30 10:43 console.php
-rw-r--r-- 1 vincent users  34K Aug 30 10:43 COPYING-AGPL
drwxr-xr-x 1 vincent users  318 Aug 30 10:44 core
-rw-r--r-- 1 vincent users 5.1K Aug 30 10:43 cron.php
-rw-r--r-- 1 vincent users  22K Aug 30 10:43 db_structure.xml
-rw-r--r-- 1 vincent users  179 Aug 30 10:43 index.html
-rw-r--r-- 1 vincent users 2.1K Aug 30 10:43 index.php
-rw-r--r-- 1 vincent users 2.6K Aug 30 10:43 indie.json
drwxr-xr-x 1 vincent users   14 Aug 30 10:43 l10n
drwxr-xr-x 1 vincent users   90 Aug 30 10:43 lib
-rw-r--r-- 1 vincent users  283 Aug 30 10:43 occ
drwxr-xr-x 1 vincent users   58 Aug 30 10:43 ocs
drwxr-xr-x 1 vincent users   42 Aug 30 10:43 ocs-provider
-rw-r--r-- 1 vincent users 3.0K Aug 30 10:43 public.php
-rw-r--r-- 1 vincent users 4.3K Aug 30 10:43 remote.php
-rw-r--r-- 1 vincent users   26 Aug 30 10:43 robots.txt
drwxr-xr-x 1 vincent users  302 Aug 30 10:43 settings
-rw-r--r-- 1 vincent users 1.8K Aug 30 10:43 status.php
drwxr-xr-x 1 vincent users   26 Aug 30 10:43 themes
-rw-r--r-- 1 vincent users  194 Aug 30 10:46 version.php
```

@cmonteroluque can you check the tarball ?

@jnweiger we might want to add a workaround in the OBS package in case the execution bit is missing.

**Comment by PVince81** [comment]
@jvillafanez can you post the update log from the package installation ? Maybe something went wrong during the update when encryption is enabled.

@jnweiger just told me that the packaging scripts already contain code to set the execution bit. Maybe something went wrong before that command could be executed.

The log should hopefully contain more information.
If possible, also check the owncloud.log from after the upgrade in case it contains information related to encryption.

**Comment by jnweiger** [comment]
chmod a+x .../occ is there since https://github.com/owncloud/core/issues/18508

**Comment by jnweiger** [comment]
@jvillafanez you say 'without encryption the execute bit is set properly' -- what Kind of encryption are you attempting?

/var/www/owncloud/occ should always be called as ```php /var/www/owncloud/occ``` and in the correct directory and as the correct user. Otherwise it misbehaves.

**Comment by jnweiger** [comment]
fix was in 8.1:testing:merged, and 8.0:testing
but it was missing in 8.1:testing 
ported over with https://build.opensuse.org/package/rdiff/isv:ownCloud:community:8.1:testing/owncloud?linkrev=base&rev=48

please try again.

**Comment by jvillafanez** [comment]
verified :+1:
